### PR TITLE
COMMERCE-9296

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-9296

@andrea-ale-sbarra

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/commerce_channel/general.jsp [hunk 1]
```diff
@@ -149,17 +149,16 @@ Map<String, String> contextParams = HashMapBuilder.<String, String>put(
...
-				<aui:select label="price-type" name="priceDisplayType">
-
-					<%
-					String priceDisplayType = commerceChannel.getPriceDisplayType();
-					%>
+				<%
+				String priceDisplayType = commerceChannel.getPriceDisplayType();
+				%>
 
+				<aui:select label="price-type" name="priceDisplayType" value="<%= priceDisplayType %>">
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/commerce_channel/general.jsp:156]: I'm not sure this is right, I mean I cannot find any example of aui:select with the value attribute, I'll try this fix locally to be sure

**Review by andrea-ale-sbarra — COMMENTED** [review]
I need to test this first

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
I checked that even in the previous version the value is correctly stored at DB so I approve this in order to make the QA verify this is the right fix

File: modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/commerce_channel/general.jsp [hunk 2]
```diff
@@ -149,17 +149,16 @@ Map<String, String> contextParams = HashMapBuilder.<String, String>put(
...
 					<aui:option label="net-price" selected="<%= priceDisplayType.equals(CommercePricingConstants.TAX_EXCLUDED_FROM_PRICE) %>" value="<%= CommercePricingConstants.TAX_EXCLUDED_FROM_PRICE %>" />
 					<aui:option label="gross-price" selected="<%= priceDisplayType.equals(CommercePricingConstants.TAX_INCLUDED_IN_PRICE) %>" value="<%= CommercePricingConstants.TAX_INCLUDED_IN_PRICE %>" />
 				</aui:select>
 
-				<aui:select label="discounts-target-price-type" name="discountsTargetNetPrice">
+				<aui:select label="discounts-target-price-type" name="discountsTargetNetPrice" value="<%= commerceChannel.getDiscountsTargetNetPrice() %>">
 					<aui:option label="net-price" selected="<%= commerceChannel.isDiscountsTargetNetPrice() %>" value="true" />
 					<aui:option label="gross-price" selected="<%= commerceChannel.isDiscountsTargetNetPrice() %>" value="false" />
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/commerce_channel/general.jsp:163]: @brianikim  the issue here is this option, the "selected" check should be
!commerceChannel.isDiscountsTargetNetPrice(), the value you are setting is just an override not the proper solution.

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]
See comment, thanks

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
@andrea-ale-sbarra , @luca-pellizzon ,

Updated commit to reflect your requests. Tested locally to confirm it works.

**Review by luca-pellizzon — APPROVED** [review]

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**